MREUnityRuntime/MREUnityRuntimeLib/Core/Actor.cs

```diff
@@ -339,58 +339,128 @@ internal void Destroy()
 			Destroy(gameObject);
 		}
 
-		internal ActorPatch GenerateInitialPatch()
+		internal ActorPatch GeneratePatch(ActorPatch output = null, TargetPath path = null)
 		{
-			LocalTransform.ToLocalTransform(transform);
-			AppTransform.ToAppTransform(transform, App.SceneRoot.transform);
-
-			var localTransform = new ScaledTransformPatch()
+			if (output == null)
 			{
-				Position = new Vector3Patch(transform.localPosition),
-				Rotation = new QuaternionPatch(transform.localRotation),
-				Scale = new Vector3Patch(transform.localScale)
-			};
+				output = new ActorPatch(Id);
+			}
 
-			var appTransform = new TransformPatch()
+			var generateAll = path == null;
+			if (!generateAll)
 			{
-				Position = new Vector3Patch(App.SceneRoot.transform.InverseTransformPoint(transform.position)),
-				Rotation = new QuaternionPatch(Quaternion.Inverse(App.SceneRoot.transform.rotation) * transform.rotation)
-			};
-
-			var rigidBody = PatchingUtilMethods.GeneratePatch(RigidBody, (Rigidbody)null, App.SceneRoot.transform);
+				if (path.AnimatibleType != "actor") return output;
+				output.Restore(path, 0);
+			}
+			else
+			{
+				output.RestoreAll();
+			}
 
-			ColliderPatch collider = null;
-			_collider = gameObject.GetComponent<UnityCollider>();
-			if (_collider != null)
+			if (generateAll || path.PathParts[0] == "transform")
 			{
-				if (Collider == null)
+				if (generateAll || path.PathParts[1] == "local")
```

Comment:
Embedding the target path schema in code like this will create a maintenance headache. We need to take a data-driven approach. You have the seed of it in TargetPath.cs: The TokenLookup class maps a value to a path. You could convert that  value to a utility class tailored to working on the specific type and value of the mapped target path.

Reply:
I don't follow. The reason this method is so bad is because the internal structure of Actor is sufficiently different from ActorPatch that there's no way to map them except explicitly, which is what I've done here.

Reply:
I want to find a solution to this that is schema-driven, so that when the schema changes, this code won't compile until it is fixed up to match the updated schema. As it is, it will be a source of bugs over time. But we can ship this as-is for now. There's a bigger clean up needed this can be a part of.